src/unit.rs

```diff
@@ -660,8 +666,21 @@ fn parse_unit_header<R: Reader>(input: &mut R) -> Result<UnitHeader<R, R::Offset
     let mut rest = input.split(unit_length)?;
 
     let version = parse_version(&mut rest)?;
-    let offset = parse_debug_abbrev_offset(&mut rest, format)?;
-    let address_size = rest.read_u8()?;
+    let offset;
+    let address_size;
+    if 2 <= version && version <= 4 {
+        offset = parse_debug_abbrev_offset(&mut rest, format)?;
+        address_size = rest.read_u8()?;
+    } else if version == 5 {
+        let unit_type = parse_compilation_unit_type(&mut rest)?;
+        if unit_type != constants::DW_UT_compile && unit_type != constants::DW_UT_type {
+            return Err(Error::UnsupportedUnitType.into());
+        }
+        address_size = rest.read_u8()?;
+        offset = parse_debug_abbrev_offset(&mut rest, format)?;
+    } else {
+        unreachable!()
```

Comment:
Delete `parse_version` and return `UnknownVersion` here instead.

Reply:
I assume you mean `unreachable!`?

Reply:
No, I meant to delete `parse_version` and replace it with `read_u16`. I don't think `parse_version` provides any value: we do a check of the version in it, then we need to immediately repeat the  check when it returns, so I think the code is clearer without `parse_version`.